server/src/calendar/continous_scraping.rs

```diff
@@ -0,0 +1,287 @@
+        (self.to + chrono::Days::new(1))
+            .signed_duration_since(self.from)
+            .num_days() as u64
+    }
+    fn split(&self) -> (Self, Self) {
+        let mid_offset = self.num_days() / 2 - 1;
+        let lower_middle = self.from + chrono::Days::new(mid_offset);
+        (
+            Self {
+                key: self.key.clone(),
+                room_id: self.room_id,
+                from: self.from,
+                to: lower_middle,
+            },
+            Self {
+                key: self.key.clone(),
+                room_id: self.room_id,
+                from: lower_middle + chrono::Days::new(1),
+                to: self.to,
+            },
+        )
+    }
+}
+
+#[cfg(test)]
+mod test_scrape_order {
+    use super::ScrapeOrder;
+    use chrono::NaiveDate;
+    #[test]
+    fn test_split() {
+        let order = ScrapeOrder::new(("".to_string(), 0), 2020, 1);
+        let (o1, o2) = order.split();
+        assert_eq!(order.from, NaiveDate::from_ymd_opt(2020, 1, 1).unwrap());
+        assert_eq!(order.to, NaiveDate::from_ymd_opt(2020, 12, 31).unwrap());
+        assert_eq!(o1.from, order.from);
+        assert_eq!(o2.to, order.to);
+        assert_eq!(o1.to + chrono::Duration::days(1), o2.from);
+    }
+    #[test]
+    fn test_split_small() {
+        let order = ScrapeOrder {
+            key: "".to_string(),
+            room_id: 0,
+            from: NaiveDate::from_ymd_opt(2020, 1, 1).unwrap(),
+            to: NaiveDate::from_ymd_opt(2020, 1, 2).unwrap(),
+        };
+        let (o1, o2) = order.split();
+        assert_eq!(o1.to + chrono::Duration::days(1), o2.from);
+        assert_eq!(order.num_days(), 2);
+        assert_eq!(order.from, o1.from);
+        assert_eq!(order.to, o2.to);
+        assert_eq!(o1.num_days(), 1);
+        assert_eq!(o2.num_days(), 1);
+        assert_eq!(order.from, o1.to);
+        assert_eq!(order.to, o2.from);
+    }
+    #[test]
+    fn test_num_days() {
+        let mut order = ScrapeOrder {
+            key: "".to_string(),
+            room_id: 0,
+            from: NaiveDate::from_ymd_opt(2020, 1, 1).unwrap(),
+            to: NaiveDate::from_ymd_opt(2020, 1, 1).unwrap(),
+        };
+        assert_eq!(order.num_days(), 1);
+        order.to = NaiveDate::from_ymd_opt(2020, 1, 2).unwrap();
+        assert_eq!(order.num_days(), 2);
+        order.to = NaiveDate::from_ymd_opt(2020, 12, 31).unwrap();
+        assert_eq!(order.num_days(), 366);
+    }
+    #[test]
+    fn test_same_day() {
+        let order = ScrapeOrder::new(("".to_string(), 0), 2020, 0);
+        assert_eq!(order.from, NaiveDate::from_ymd_opt(2020, 1, 1).unwrap());
+        assert_eq!(order.to, NaiveDate::from_ymd_opt(2019, 12, 31).unwrap());
+        assert_eq!(order.num_days(), 0);
+    }
+}
+
+async fn scrape(
+    client: &Client,
+    id: (String, i32),
+    from_year: i32,
+    year_duration: i32,
+) -> Result<usize, usize> {
```

Comment:
Do I understand it correctly, that `Err` is only used to indicate that a retry was necessary, but not that the scraping failed? Maybe a custom `ScrapeResult` struct would be better then

Reply:
fully agree